packages/x-charts/src/internals/components/ChartsText.tsx

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react';
+import { getStringSize } from '../domUtils';
+
+interface GetWordsByLinesParams {
+  /**
+   * Text displayed.
+   */
+  text: string;
+  /**
+   * Style applied to text elements.
+   */
+  style?: React.SVGAttributes<'text'>['style'];
+  /**
+   * If true, the line width is computed.
+   * @default false
+   */
+  needsComputation?: boolean;
+}
+
+export type ChartsTextBaseline = 'hanging' | 'central' | 'auto';
+
+export interface ChartsTextProps
+  extends Omit<React.SVGTextElementAttributes<SVGTextElement>, 'width' | 'ref'>,
+    GetWordsByLinesParams {
+  /**
+   * Height of a text line (in `em`).
+   */
+  lineHeight?: number;
+  /**
+   * The text baseline
+   * @default 'central'
+   */
+  dominantBaseline?: ChartsTextBaseline;
```

Comment:
I don't understand. Previously the text was with `central`. Does `middle` have a specific advantage?

![image](https://github.com/mui/mui-x/assets/45398769/ebd3b188-e960-4493-962b-b41e4e343317)

I took a screenshot of the preview and production, and I struggle to find the breaking change

![image](https://github.com/mui/mui-x/assets/45398769/1837b85b-7da8-4806-b2f2-4227ab3144de)

Reply:
Fair point. 🙈 
My bad, it's more of a general issue that we can leave for a separate PR if need be. 👌 

If I understand it correctly, the problem seems to stem from the fact that the legend label appears misaligned vertically, event though it is aligned, but based on the tallest letter.
WDYT about changing all the demo legend labels to start with a capital letter in order to avoid this strange visual anomaly? 🙈 🤷 🤔 
![Screenshot 2023-10-10 at 20 04 43](https://github.com/mui/mui-x/assets/4941090/3219b556-bd59-4bc3-9492-0882f28d5f86)

Reply:
Ok, now I remember why we moved from `middle` to `central`. It was for the tick alignment. Digits looked better aligned with `central`

Good idea to add first captial letter